src/app/services/skill-service.service.ts

```diff
@@ -0,0 +1,45 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { Skill } from '../models/skill';
+import { Observable } from 'rxjs';
+
+
+@Injectable()
+export class SkillService {
+
+private skillsUrl: string;
+private saveSkillUrl: string;
+private deleteSkillUrl: string;
+private relinkSkillsUrl: string;
+
+private  returnval : Observable<Skill[]>;
+private uskills: Skill[];
+private    pskills: Observable<Skill[]> ;
+
+  constructor(private http: HttpClient) {
+    this.skillsUrl = 'http://localhost:8080/getskills';
+    this.saveSkillUrl = 'http://localhost:8080/saveskill';
+    this.deleteSkillUrl = 'http://localhost:8080/deleteskill';
+    this.relinkSkillsUrl = 'http://localhost:8080/relinkskills';
```

Comment:
Please do not hardcode any url like this. Use the environment variables for this.

```typescript
import { environment } from 'src/environments/environment';
```

Change line 27 to this (as an example):
```typescript
return this.http.get<Skill[]>(environment.api + '/getskills');
```

Reply:
adapted